Undefine NDEBUG for tests

Our build system defines `-DNDEBUG` via `CPPFLAGS` for release builds. Because of this, the tests don't compile. 

```
  CC       test2.o                                                                                                                                                                                        [21/1905]
  CCLD     test2
  CC       test4.o
  CCLD     test4
  CC       testReplaceExisting.o
  CCLD     testReplaceExisting
  CC       test_parse_int64.o
  CCLD     test_parse_int64
  CC       test_deep_copy.o
test_deep_copy.c: In function 'main':
test_deep_copy.c:201:14: error: unused variable 'special_output' [-Werror=unused-variable]
  const char *special_output = json_object_to_json_string(dest_with_serializer);
              ^~~~~~~~~~~~~~
test_deep_copy.c:198:8: error: unused variable 'dst_userdata' [-Werror=unused-variable]
  char *dst_userdata = json_object_get_userdata(dest_with_serializer);
        ^~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [test_deep_copy.o] Error 1
make[2]: Leaving directory `/home/nwani/m3/conda-bld/json-c_1520910223423/work/tests'
make[1]: *** [check-am] Error 2
make[1]: Leaving directory `/home/nwani/m3/conda-bld/json-c_1520910223423/work/tests'
```

I think all all tests should have:
```
#ifdef NDEBUG
#/**/undef/**/ NDEBUG
#endif
```

But I couldn't find a header file common to only tests.

--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -1,5 +1,5 @@
 
-AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/tests
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/tests -UNDEBUG
 
 LDADD= $(LIBJSON_LA)
 

